modules/renter/hostdb/scan.go

```diff
@@ -110,9 +113,16 @@ func (hdb *HostDB) threadedProbeHosts() {
 			defer hdb.mu.Unlock()
 
 			// Regardless of whether the host responded, add it to allHosts.
-			if _, exists := hdb.allHosts[hostEntry.NetAddress]; !exists {
+			priorHost, exists := hdb.allHosts[hostEntry.NetAddress]
+			if !exists {
 				hdb.allHosts[hostEntry.NetAddress] = hostEntry
 			}
+			if exists && !bytes.Equal(priorHost.PublicKey.Key, hostEntry.PublicKey.Key) {
+				// The public key in the announcement doesn't match the public
+				// key we have locally, and the host did not respond - can't be
+				// sure we have the right host, don't decrement reliability.
+				return
+			}
```

Comment:
comments says `host did not respond`, but there's no `err != nil` here. Seems like this won't add ever add a host to `activeHosts` if the keys mismatch.